app/src/main/java/com/example/habitsmasher/ui/dashboard/EditHabitFragment.java

```diff
@@ -98,6 +112,110 @@ public void onClick(View v) {
             }
         });
 
+        //setting day initial states (clicked or not)
+        if (_tracker.getMonday()){mondayButton.performClick();}
+        if (_tracker.getTuesday()){tuesdayButton.performClick();}
+        if (_tracker.getWednesday()){wednesdayButton.performClick();}
+        if (_tracker.getThursday()){thursdayButton.performClick();}
+        if (_tracker.getFriday()){fridayButton.performClick();}
+        if (_tracker.getSaturday()){saturdayButton.performClick();}
+        if (_tracker.getSunday()){sundayButton.performClick();}
```

Comment:
These set the initial cases of the days, whether checked or not.

Reply:
This is also something that can be extracted to its own helper method! Try to look for seams in the code that deviate from what the original method is trying to do, and extract them out to be more readable.

Here, we can take all of these lines and stick them in a method called `setInitialStatesForWeeklySchedule()` or something similar.